### ISSUE TITLE
Crafting priority

### ISSUE THREAD

**Issue by gixbit** [AppliedEnergistics/Applied-Energistics-2#1055]
I am posting this knowing an issue was made in the past over this but, Theres a couple things I'm unsure of. 

One thing is the actual priority in which certain crafting units are utilized. Because, mine always utilizes my 392k crafting storage with 72 co-processors but, the 4 smaller ones with 32k storage and 20 co-processors is hardly used, IF ever, from my inspections anyway. 

The other is how do I actually utilize the current mechanic the way I want.

I am not sure how you calculate your current priorities however, Would it work to prioritize by two completely workable things. First being how many items need to be created, the second being how much of it needs to be made. 

So if the amount items being made which will usually be one for autocrafting will be factored into how much space. In light of this, it's immediately obvious that storage capacity is not the biggest factor since even the smallest capacities can handle single unit jobs, except when recipes require a stack of an output or input. So, it is still a scalable factor.

Lastly, crafting operations should also be factored in my opinion which is purely my own opinion here since I don't know what this involves. How hard would it be to calculate how many items currently need to be created to achieve finalization? Then utilize the CPU with the appropriate number of co-processors. So if you're making a stick, Make it so that it uses the smallest storage,co-processor CPU. If you're making resonant pulverizers, maybe it uses one with 20 co-processors vs. the one with 72.

**Comment by yueh** [comment]
Mostly impossible, not from the code itself but based on each players intent. Some want just the have their item the fastest possible way, some will not really care but a probably happy when it is faster and the ones actually wanting to use the perfect cpu are not a large portion (and probably also enjoy the fastest option sometimes)

If I remember correctly, there is even no information available on how many assemblers are attached to the same interface. But even it that information would be available, consider this scenario:
You want to craft something with multiple intermediate products, which alle can use different amounts of co procs. Say one item can be crafted in 8 different interfaces, one in 4 and the other 2 just in a single one. 
Do you now pick the CPU with 7 co procs or one with 3 or none?
Would you consider using the one with 7 as wasting because it will be only used for a small portion of the request, the one with none, or is something in between the best for you personally? (and what would another player assume as best?)
Also we have no idea about the time a crafting operation takes, so not taking the one with 7 co procs might take 8x times as long or the choice will not affect it at all as the items with just 1 interface will take the majority of the time.

Basically any solution will make some people not happy.
So just sticking with use the fastest available CPU does probably cover the most cases.

**Comment by gixbit** [comment]
So thats how it decides? It picks the fastest available CPU from the connected CPUs. That makes some varying amount of sense. I guess if I want to make my CPUs free, I need to plan it out better.

**Comment by thatsIch** [comment]
If you craft it manuallly, it will always pick the fastest CPU. 
Automatic crafting through export bus and interface are done via the slowest ones.

**Comment by gixbit** [comment]
That so? I never seen it use anything but the fastest one even automatic crafting. Are interfaces handled the same with automatic crafting priority? I'm not inclined to tell you you are wrong since you guys wrote the program, I'll have to re-examine it myself to find out I am actually wrong.

**Comment by thatsIch** [comment]
By fastest, I meant biggest, since there is no real estimation for a C-CPU being "fast". Automatic crafting will use the C-CPU with the least required bytes capactiy.

If not, then its a bug.

**Comment by gixbit** [comment]
No you're right, It does work that way. I made two big crafters and 4 smaller ones with the purpose of using a mechanic that I guessed would be in place. Turns out I am wrong. I'm happy to be wrong since this makes my life easier.
